Update neoforge to the latest version & Ars Nouveau compatibility

QianMo0721 opened the pull request:
Bump Neoforge version to 21.1.169
Ars Nouveau compatibility: Fixed a bug where the server crashed and threw "Fix Cannot invoke 'io.izzel.arclight.common.mod.util.EntityDamageResult.armorDamageOffset()' because 'this.entityDamageResult0' is null" when the player cast a spell

QianMo0721 committed:
Bump Neoforge version to 21.1.169

QianMo0721 committed:
Ars Nouveau compatibility

Fixed a bug where the server crashed and threw "Fix Cannot invoke 'io.izzel.arclight.common.mod.util.EntityDamageResult.armorDamageOffset()' because 'this.entityDamageResult0' is null" when the player cast a spell

QianMo0721 committed:
Bump Neoforge version to 21.1.170

InitAuther97 commented:
Actually, the damage calculation works basically like a bunch of shit on NeoForge and won't calculate the damage correctly for Bukkit events. It will be nice if you can look into it and provide a better solution.

QianMo0721 commented:
Okay, I will make time to study it. Although my development experience is not much : D



---Original---
From: ***@***.***&gt;
Date: Tue, Jun 3, 2025 18:34 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [IzzelAliz/Arclight] Update neoforge to the latest version &amp; ArsNouveau compatibility (PR #1813)


InitAuther97 left a comment (IzzelAliz/Arclight#1813)
 
Actually, the damage calculation works basically like a bunch of shit on NeoForge and won't calculate the damage correct for Bukkit event. It will be nice if you can look into it and provide a better solution.
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***&gt;

QianMo0721 committed:
fix(installer): Neoforge will be reinstalled every time the server is restarted

Modified the forgeClasspathMissing method in NeoForgeInstaller so that it only checks critical files:
Only check if the args.txt file exists
Only the key documents related to neoforge are checked
Ignore checks for other non-critical files
Added log output
